src/Domain/LeanCode.DomainModels.MassTransitRelay/Middleware/EventsPublisherElement.cs

```diff
@@ -44,25 +41,18 @@ private Task PublishEventsAsync(TContext ctx, List<IDomainEvent> events)
         return Task.WhenAll(publishTasks);
     }
 
-    [System.Diagnostics.CodeAnalysis.SuppressMessage(
-        "?",
-        "CA1031",
-        Justification = "The method is an exception boundary."
-    )]
-    private async Task PublishEventAsync(IDomainEvent evt, TContext ctx, Guid conversationId)
+    private Task PublishEventAsync(IDomainEvent evt, TContext ctx, Guid conversationId)
     {
         logger.Debug("Publishing event of type {DomainEvent}", evt);
-
-        try
-        {
-            await publisher.PublishAsync(evt, conversationId, ctx.CancellationToken);
-        }
-        catch (Exception e)
-        {
-            logger.Fatal(e, "Could not publish event {@DomainEvent}", evt);
-        }
-
-        logger.Information("Domain event {DomainEvent} published", evt);
+        return publishEndpoint.Publish(
+            (object)evt,
```

Comment:
Why cast here?

Reply:
Without a cast you'll get `Publish<IDomainEvent>` overload which will publish the event as `IDomainEvent` - not the one you expect.

Object overload will use a runtime type - though a comment would be nice.